### DISCUSSION TITLE
Bagaimana Cara Membuat Feed Kustom di Mecha?

### DISCUSSION THREAD

**Discussion by yuliakbard** [mecha-cms/mecha#180] [Q&A]
Feed bawaan mecha `feed.xml` atau `feed.json`, yang mana pada bagian `<description></description>` menggunakan `$page->description`. saya ingin menggunakan `$page->content` pada bagian `<description></description>` tanpa harus mengubah feed bagian dari `feed.xml`, untuk penggunaan instant article facebook atau google news publications.

karena saya ingin menghindari pencurian content melalui feed. Mohon jawabannya kang.

Sejujurnya saya kepikiran untuk menduplikat extension feed trus di custom, tapi apakah ada opsi lain?

**Comment by yuliakbard** [comment]
Dan apakah bisa ditambahkan http authetication setiap ada yang mencoba mengakses url feed kita akan menampilkan alert login username dan password seperti feed pada blogger?

**Reply by taufik-nurrohman** [reply]
https://www.php.net/manual/en/features.http-auth.php

**Comment by taufik-nurrohman** [comment]
Coba begini:

~~~ php
if ('feed.xml' === basename($url->path)) {
    Hook::set('page.description', function() {
        return $this->content;
    }, 0);
}
~~~
